### PR TITLE
scheduler: improve gang log

### PR DESCRIPTION
**Pull request by googs1025**
### Ⅰ. Describe what this PR does
I modified the current log level output method. In order to reduce log printing, I only print when --v=4 is used, and use klog.V().ErrorS for the err!=nil part.

### Ⅱ. Does this pull request fix one issue?
fix: https://github.com/koordinator-sh/koordinator/issues/1895#issuecomment-2060218803

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

**Comment by hormes** [comment]
Why do I need to downgrade error logs to V4?

**Comment by googs1025** [comment]
> Why do we need to downgrade error logs to V4?

Indeed, this level needs to be confirmed. However, setting the log level can effectively reduce the number of log prints.

File: pkg/scheduler/plugins/coscheduling/core/gang.go [hunk 1]
```diff
@@ -145,7 +146,7 @@ func (gang *Gang) tryInitByPodConfig(pod *v1.Pod, args *config.CoschedulingArgs)
 	matchPolicy := util.GetGangMatchPolicyByPod(pod)
 	if matchPolicy != extension.GangMatchPolicyOnlyWaiting && matchPolicy != extension.GangMatchPolicyWaitingAndRunning &&
 		matchPolicy != extension.GangMatchPolicyOnceSatisfied {
-		klog.Errorf("pod's annotation AnnotationGangMatchPolicy illegal, gangName: %v, value: %v",
+		klog.V(4).Infof("pod's annotation AnnotationGangMatchPolicy illegal, gangName: %v, value: %v",
```
Comment (eahydra) [inline pkg/scheduler/plugins/coscheduling/core/gang.go:149]: We should determine if matchPolicy is an empty string and if not, log an error.

File: pkg/scheduler/plugins/coscheduling/core/gang.go [hunk 2]
```diff
@@ -124,19 +124,20 @@ func (gang *Gang) tryInitByPodConfig(pod *v1.Pod, args *config.CoschedulingArgs)
...
 			gang.Name, pod.Annotations[extension.AnnotationGangTotalNum])
 		totalChildrenNum = int64(minRequiredNumber)
 	} else if totalChildrenNum != 0 && totalChildrenNum < int64(minRequiredNumber) {
-		klog.Errorf("pod's annotation totalNumber cannot less than minRequiredNumber, gangName: %v, totalNumber: %v,minRequiredNumber: %v",
+
+		klog.V(4).Infof("pod's annotation totalNumber cannot less than minRequiredNumber, gangName: %v, totalNumber: %v,minRequiredNumber: %v",
 			gang.Name, pod.Annotations[extension.AnnotationGangTotalNum], minRequiredNumber)
 		totalChildrenNum = int64(minRequiredNumber)
 	}
 	gang.TotalChildrenNum = int(totalChildrenNum)
```
Comment (eahydra) [inline pkg/scheduler/plugins/coscheduling/core/gang.go:136]: We encourage cleaner code, which of course includes improved logging.

```suggestion
var expectedTotalNum int64
s := pod.Annotations[extension.AnnotationGangTotalNum]
if s != "" {
   var err error
   expectedTotalNum, err = strconv.ParseInt(s)
   if err != nil {
      klog.V(4).ErrorS(err, "illegal annotation", "key", extension.AnnotationGangTotalNum, "value": s)
   }
}
if expectedTotalNum < int64(minRequiredNumber) {
   expectedTotalNum = int64(minRequiredNumber)
}
gang.TotalChildrenNum = int(expectedTotalNum)
```

File: pkg/scheduler/plugins/coscheduling/core/gang.go [hunk 3]
```diff
@@ -157,15 +158,15 @@ func (gang *Gang) tryInitByPodConfig(pod *v1.Pod, args *config.CoschedulingArgs)
 
 	waitTime, err := time.ParseDuration(pod.Annotations[extension.AnnotationGangWaitTime])
 	if err != nil || waitTime <= 0 {
-		klog.Errorf("pod's annotation GangWaitTimeAnnotation illegal, gangName: %v, value: %v",
+		klog.V(4).ErrorS(err, "pod's annotation GangWaitTimeAnnotation illegal, gangName: %v, value: %v",
```
Comment (eahydra) [inline pkg/scheduler/plugins/coscheduling/core/gang.go:161]: All these logs have same problems, should improve these codes to make more clean.

**Comment by googs1025** [comment]
Sorry for being late. I've been busy with other work recently. I'll go through the comments and make some revisions during the holidays.

**Review by hormes — APPROVED** [review]

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Pull request merged**